whitepapers/0010_ibc_generic_messaging

```diff
@@ -0,0 +1,92 @@
+
+### IBC Relayers
+
+All IBC communication is facilitated by [IBC relayers](https://ibcprotocol.org/relayers/). Since these are lightweight processes that need to only listen to blockchain RPC nodes, each (only several is also acceptable) wormhole guardian can run a relayer.
+
+The guardian IBC relayers are configured to connect the `wormchain-ibc-receiver` contract on wormchain to the various `wormhole-ibc` contracts on the cosmos chains that wormhole supports.
+
+### Guardian Node Watcher
+
+We will modify the cosmos guardian watcher to watch the `wormchain-ibc-receiver` contract on wormchain for the messages from the designated `wormhole-ibc` contracts on supported cosmos chains. This is nearly identical to the current model and can be a drop in replacement.
+
+The new guardian watcher can verify that messages originate from the chain they claim to originate from by checking the IBC connection ID. The `wormchain-ibc-receiver` contract logs the connection ID that the message was received over. Since IBC connections are unique and can never be updated or closed ([docs](https://github.com/cosmos/ibc/tree/main/spec/core/ics-003-connection-semantics#sub-protocols)), the guardians can associate connection IDs with specific chains.
+
+### API / database schema
+
+```rust
+/// This is the message we send over the IBC channel
+#[cw_serde]
+pub enum WormholeIbcPacketMsg {
+    Publish { msg: Response }
+}
+```
+
+## Deployment
+
+There are several steps required to deploy this feature. Listed in order:
+
+1. Deploying the new contracts: `wormhole-ibc` contracts to Cosmos chains and the `wormhole-ibc-receiver` contract to Wormchain.
+2. Upgrading existing `wormhole` contracts on Cosmos chains to use the new `wormhole-ibc` bytecode.
+3. Establishing IBC connections between the `wormhole-ibc` contracts and the `wormhole-ibc-receiver` contract.
+4. Upgrading the guardian software.
+
+First, we need to deploy the `wormhole-ibc-receiver` contract on Wormchain. This will require 2 governance VAAs to deploy and instantiate the bytecode.
+
+Once we know the `wormhole-ibc-receiver` contract address, we can hardcode this (or, using an environment variable during compilation) into the `wormhole-ibc` contracts. Recall that these need to know the address of the `wormhole-ibc-receiver` contract to look up the correct IBC channel over which to send messages.
+
+Next, we should deploy the compiled `wormhole-ibc` contracts to Cosmos chains we already support (Terra2, XPLA, and Injective). We can migrate the existing `wormhole` contract to the new `wormhole-ibc` bytecode so that we don't need to redeploy and re-instantiate the Cosmos token bridge contracts with new core bridge contract addresses.
```

Comment:
```suggestion
Next, we should deploy the compiled `wormhole-ibc` contracts to ibc enabled chains we already support (Terra Classic, Terra2, XPLA, and Injective). We can migrate the existing `wormhole` contract to the new `wormhole-ibc` bytecode so that we don't need to redeploy and re-instantiate the token bridge contracts with new core bridge contract addresses.
```